### ISSUE TITLE
How to limit the scope of the profiler?

### ISSUE THREAD

**Issue by hyhuang00** [pytorch/kineto#615]
I am wondering if it is possible to limit the scope of the profiler to a particular part of the neural network. Currently, I am trying to analyze the bottleneck of my model using the following pseudocode:

```
import torch.profiler as profiler
    with profiler.profile(
            activities=[
                profiler.ProfilerActivity.CPU,
                profiler.ProfilerActivity.CUDA,
            ],
            profile_memory=True,
            schedule=profiler.schedule(wait=5, warmup=2, active=1, repeat=1), 
            on_trace_ready=profiler.tensorboard_trace_handler(tensorboard_logdir)
    ) as p:
    for sample in dataloader:
         model(sample)
```

However, the trace I created is still way too large (~800MB) for the tensorboard to function properly. Apparently tensorboard is only able to load the trace if it is smaller than about 500 MB, so I am thinking about limiting the trace of the profiler to only look at part of the neural net that leads to the issue. However, it seems like a warmup is necessary, so inserting the profiler.profile within a network will generate inaccurate results. Is there a way to limit the scope of the profiler without breaking the interface?

**Comment by aaronenyeshi** [comment]
Since `p.step()` is used to mark iteration boundaries, you may try using that to split up the model. Other than that, you should be able to visualize an 800MB trace on chrome using export_chrome_trace. If the trace starts going over 5GB, you can try using Perfetto to visualize the trace file.

**Comment by hyhuang00** [comment]
Thank you for your help!

- I tried to visualize the trace on chrome via "chrome://tracing" but this method works only when I got the trace smaller than 500MB as well. My machine has a 32GB memory so I think the memory is sufficient, Is there a particular parameter I should modify to allow larger memory usage? 
- For the Perfetto trick, I tried either upload my trace directly or use `trace_processor` to process the trace. Both tricks don't work for me -- what I got on the Perfetto UI is just blank. Do you know any solutions to this problem?
- Using p.step() to define is doable, but my network include some conditional computation, so I cannot use the regular schedule trick to ensure I always capture the correct module. Besides, if I use another module to do the warmup, will that lead to inaccurate results?

**Comment by aaronenyeshi** [comment]
For the Perfetto trick, we have an internal script for this (ping me internally if you are an employee). Otherwise, you need to manually modify the .json file, by moving the traceEvents: [] structure to be at the root of the json. You can also delete the other root level json entries in the file. Ie:
```
{
  "schemaVersion": 1,
  "deviceProperties": [
    ...
  ],
  "traceEvents": [
   {
     ... <op>
   },
   <more ops>
  ],
  ...
}
```
into:
```
[
   {
     ... <op>
   },
   <more ops>
]
```
Another alternative is on-demand tracing via time duration. Use a duration of 300ms or less. (FYI - this is only internal)
